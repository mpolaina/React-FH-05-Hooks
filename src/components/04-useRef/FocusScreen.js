import React, { useRef } from 'react'
import logo from '../../logo.svg'

export const FocusScreen = () => {
  
  const inputRef = useRef()
  console.log(inputRef)
  
  const handleClick = () => {
    inputRef.current.select()
  }
  
  return (
    
    <div className="p-5">
      <h2 className="fw-bold"> <img src={ logo } alt="" className="logo"/>Focus Screen</h2>
      <div className="form-group">
          <label className="form-label fw-bold">Nombre</label>
          <input
            ref={ inputRef } 
            className="form-control w-50"
            placeholder="Su nombre"
            type="text"
          />
      </div>
      <button 
        className="btn btn-outline-primary mt-3"
        onClick={ handleClick }
      >Focus</button>
    </div>
  )
}
