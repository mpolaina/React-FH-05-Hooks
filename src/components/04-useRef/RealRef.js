import React, { useState } from 'react'
import logo from '../../logo.svg'
import { MultipleCustomHooks} from '../03-examples/MultipleCustomHooks'

export const RealRef = () => {
  
  const [show, setShow] = useState(false)
  
  return (
    <div className="p-5">
      <h2 className="fw-bold">
        <img src={ logo } alt="" className="logo"/>
        Real Example useRef
      </h2>
      
      <button
        className="btn btn-primary mt-3"
        onClick={ () => setShow( !show ) }
      >Show/Hide
      </button>
      
      { show && <MultipleCustomHooks/> }
      
      
    </div>
  )
}
