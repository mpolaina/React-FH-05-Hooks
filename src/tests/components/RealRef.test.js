import { shallow } from "enzyme"
import { RealRef } from "../../components/04-useRef/RealRef"

describe('Pruebas en RealRef', () => {
  
    const wrapper = shallow(<RealRef/>)
    
    test('Debe mostrarse correctamente', () => {
    
      expect( wrapper ).toMatchSnapshot()
      expect( wrapper.find('MultipleCustomHooks').exists()).toBe(false)
    })
    
    test('Debe mostrar el componente MultipleCustomHook', () => {
      
      wrapper.find('button').simulate('click')
      expect( wrapper.find('MultipleCustomHooks').exists()).toBe(true)
    })
    
    
})