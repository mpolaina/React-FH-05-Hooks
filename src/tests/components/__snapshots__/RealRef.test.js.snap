// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealRef Debe mostrarse correctamente 1`] = `
<div
  className="p-5"
>
  <h2
    className="fw-bold"
  >
    <img
      alt=""
      className="logo"
      src="logo.svg"
    />
    Real Example useRef
  </h2>
  <button
    className="btn btn-primary mt-3"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
